pkg/model/manifest.go

```diff
@@ -460,6 +462,8 @@ var localTargetAllowUnexported = cmp.AllowUnexported(LocalTarget{})
 var selectorAllowUnexported = cmp.AllowUnexported(container.RefSelector{})
 var refSetAllowUnexported = cmp.AllowUnexported(container.RefSet{})
 var registryAllowUnexported = cmp.AllowUnexported(container.Registry{})
+var ignoreCustomBuildDepsField = cmpopts.IgnoreFields(CustomBuild{}, "Deps")
```

Comment:
I'm a little bit nervous about putting these options in a generic DeepEqual function. it seems very likely to bite someone and lead to other bugs.

can we pass these cmpopts in as an option from ChangesInvalidateBuild?

Reply:
(i would also be fine with changing the name of the function and all its callers to express that it's only checking for build invalidation)